src/main/java/betterquesting/api/utils/NBTConverter.java

```
@@ -292,21 +292,26 @@ public static NBTTagCompound JSONtoNBT_Object(JsonObject jObj, NBTTagCompound ta
             if (!format) {
                 tags.setTag(key, JSONtoNBT_Element(entry.getValue(), (byte) 0, false));
             } else {
-                String[] s = key.split(":");
+                // Optimized key parsing without String.split()
                 byte id = 0;
+                String keyToUse = key;
 
                 try {
-                    id = Byte.parseByte(s[s.length - 1]);
-                    key = key.substring(0, key.lastIndexOf(":" + id));
-                } catch (Exception e) {
+                    int lastColonIndex = key.lastIndexOf(':');
+                    if (lastColonIndex != -1) {
+                        id = Byte.parseByte(key.substring(lastColonIndex + 1));
+                        keyToUse = key.substring(0, lastColonIndex);
```

shouldn't this be `key.lastIndexOf(":" + id)`? Also, i think the `lastColonIndex != -1` should get removed and the try-catch should catch an Exception, so that if it can't find the index it'll get logged instead of silently failing